### PR TITLE
Migrate eslint-config

### PR DESCRIPTION
**Pull request by aht007**
JIRA Issue: https://openedx.atlassian.net/browse/BOM-3382

## DESC
This PR replaces the deprecated `eslint-config-edx` with the newer `@edx/eslint-config` package.
Most of the changes in this PR are related to linting because of the newer configuration plugin. These changes couldn't be split into multiple PRs as current changes are not compatible with older plugin and hence all the changes need to be catered in this PR.

### CODE REVIEW

File: src/components/programs-list/tests/__snapshots__/ProgramListPage.test.jsx.snap [hunk 1]
```diff
@@ -836,7 +822,7 @@ exports[`ProgramListPage renders fetching program error page when there are issu
         <p>
           © 
           2022
-           edX Inc. All rights reserved.
```
Comment (aht007) [inline src/components/programs-list/tests/__snapshots__/ProgramListPage.test.jsx.snap:839]: 2 tests were failing after migrating the `eslint` package and I had to update the snapshots to get those tests passing. These changes to the snapshot look trivial to me.

**Comment by schenedx** [comment]
@aht007 Thank you for putting this together. Looked like a lot of work.

1. Looks like the commit lint is broken. Could you please investigate a bit and find out if this only affect this repo, or all repos?
2. There are a lot of format and keywords changes with this PR. Are you planning to run the code on local machine to run manual testing? Or you are going to rely on staging environment for validation?
3. The changes on upgrading packages looks good.

**Comment by aht007** [comment]
@schenedx I will run the MFE locally tomorrow for validating the changes as well as look into the commit lint failure as well.

**Comment by aht007** [comment]
@schenedx Can you please list out the steps on how to test this MFE locally...? I couldn't find it in the README file and when accessing the MFE locally I see the following message
Screen Shot 2022-04-28 at 5 23 51 PM

**Comment by schenedx** [comment]
@aht007 I think by seeing that message, it's good enough. Setting up to test full experience on this repository is very involved across 3 different repos (edx-platform, discovery and portal-designer). Feel free to merge and deploy to stage. Test on stage will help to validate more.

**Review by awais786 — APPROVED** [review]

**Pull request merged**